src/cat.sol

```diff
@@ -117,28 +128,42 @@ contract Cat is LibNote {
     }
 
     // --- CDP Liquidation ---
-    function bite(bytes32 ilk, address urn) external returns (uint id) {
-        (, uint rate, uint spot) = vat.ilks(ilk);
-        (uint ink, uint art) = vat.urns(ilk, urn);
+    function bite(bytes32 ilk, address urn) external returns (uint256 id) {
+        (, uint256 rate, uint256 spot) = vat.ilks(ilk);
+        (uint256 ink, uint256 art) = vat.urns(ilk, urn);
 
         require(live == 1, "Cat/not-live");
         require(spot > 0 && mul(ink, spot) < mul(art, rate), "Cat/not-unsafe");
+        require(litter < box, "Cat/liquidation-limit-hit");
+
+        Ilk memory milk = ilks[ilk];
+
+        uint256 limit = min(milk.lump, sub(box, litter));
+        uint256 fart = min(art, mul(limit, RAY) / rate / milk.chop);
```

Comment:
imo you shouldn't divide by `chop` here as it represents an additional penalty factor on top of the debt that is being covered, rather than any additional debt. `chop` is more of an auction implementation detail and actually was nearly moved to the flipper.

Reply:
I considered this, but since we're trying to limit the amount of Dai out for liquidation and the `chop` is technically part of that Dai out for liquidation.  In theory, at 13% across multiple collateral types and a 10 million box or larger we could be looking at an additional 1.3 million or more of Dai demand for liquidations.  I think, if this is a consideration that just breaks our theoretical model, that we should still try to capture this Dai in the `box`.

Do you see this as a fundamental break in the accounting or as a break in the theoretical model if how we apply `chop`?  I'm trying to figure out if we should claw this back and perhaps leave the `chop` out of litterbox accounting, or not.  For now, if this just breaks our mental model of `chop` as a feature of auctions, I think we should account for it in the litterbox.  As always, interested in your thoughts.

Reply:
`chop` only increases the threshold at which the two-way auction flips and begins to return collateral to the cdp owner. Whether the auction reaches that threshold or not depends on the market price. Similarly, the `lot` doesn't exactly represent the "collateral out to auction" as in the reverse auction this amount is reduced. 

Yes, I think `chop` should be left out of the `box`. The `box` would then represent exactly the total cdp debt being covered by liquidation, and it is easy for governance to set it to $10m when they know that really they want the maximum to be $13m, knowing that the penalty is 13%. The accounting isn't really broken otherwise, although I'm wondering about rounding errors, but it doesn't feel great to me that `chop` is part of the calculation for an argument to `grab`.

Reply:
`claw` is a great meme btw :p, maybe even more apt than `scoop`

Reply:
@rainbreak the original motivation for including `chop` in `box` was so that when the auction concludes and `claw` is called, the code doesn't have to correct for `chop` (the auction only knows `art * rate * chop`), which might have changed since the auction began, and would itself be subject to rounding error.

I'm revisiting this design decision right now since it would significantly simplify `bite` and probably governance's reasoning about `box` to leave `chop` out (and reduce precision loss, although I don't think that's a big issue with the current design). The trade-off is that in order to deal with the problem I mentioned above (`chop` can change and you can't perfectly back-calculate even if it hasn't), it's necessary to store the pre-`chop` `tab` value somewhere, either in the `Cat` (indexed by auction id) or in the `Flipper`. Any opinions about that trade-off?

Reply:
Another option is to move `chop` to the flipper. Or if that's too drastic a change, add a `chop` argument to `kick`, along with the `tab`.

Reply:
I thought about putting `chop` in the Flipper (or passing it), but I like keeping the concerns separated--the Flipper just auctions collateral, and the Cat worries about liquidation details like the penalty. I also like not making changes to the Flipper's public APIs (even modest ones) since it's a focus point for integrations.

I have a PR for this over here if you're interested: https://github.com/makerdao/dss/pull/128